assets/js/bulk-move.js

```diff
@@ -8,10 +8,46 @@
 
 /*jslint browser: true, devel: true*/
 /*global BULK_MOVE, jQuery, document, postboxes, pagenow, ajaxurl*/
+BULK_MOVE.validate_same_user_roles = function(that) {
+
+    var fromUserRole = jQuery( '#bm-users-by-role-from-roles-list' ).val().toLowerCase(),
+        ToUserRole = jQuery( '#bm-users-by-role-to-roles-list' ).val().toLowerCase();
+
+    fromUserRole = jQuery.trim( fromUserRole );
+    ToUserRole = jQuery.trim( ToUserRole );
+
+    return fromUserRole === ToUserRole;
+};
+
 jQuery(document).ready(function () {
-	jQuery( 'button[value="move_tags"], button[value="move_cats"], button[value="move_category_by_tag"], button[value="move_custom_taxonomy"]' ).click( function () {
-		return confirm( BULK_MOVE.msg.move_warning );
-	});
+
+    jQuery('button[value="move_tags"], button[value="move_cats"], button[value="move_category_by_tag"], button[value="move_custom_taxonomy"], button[value="move_users_by_role"]').click(function (e) {
+
+        var currentButton = jQuery(this).val(),
+            valid = true,
+            that = this,
+            errorKey;
+
+        if (!currentButton in BULK_MOVE.validators) {
```

Comment:
You are using a negation in the lhs of this in expression. Are you sure you did not want to negate the result of the expression like so `! (currentButton in BULK_MOVE.validators)`?

Reply:
> You are using a negation in the lhs of this in expression. Are you sure you did not want to negate the result of the expression like so `! (currentButton in BULK_MOVE.validators)`?

You're right. It worked in this case, because `!` operator reads right-to-left. However, I added parenthesis.

Reference:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence